pkg/netconfig/netconfig.go

```diff
@@ -208,6 +235,22 @@ func UpdateIPSet(ipVersion, ipSetName, egressFilterList, defaultNetworkDevice st
 	return nil
 }
 
+func RemoveIPSet(ipVersion, ipSetName, defaultNetworkDevice string) error {
+	err := RemoveIPTablesLoggingRules(ipVersion, ipSetName, defaultNetworkDevice)
+	if err != nil {
+		return fmt.Errorf("error removing iptables rules %w", err)
+	}
+
+	if err := DefaultNetUtilsCommandExecutor.ExecuteIPSetCommand("list", ipSetName); err == nil {
+		err = DefaultNetUtilsCommandExecutor.ExecuteIPSetCommand("destroy", ipSetName)
+		if err != nil {
+			return fmt.Errorf("error cleaning-up ipset %s: %w\n", ipSetName, err)
+		}
+	}
+
+	return err
```

Comment:
Here and in other places you diverge from the default error handling. Why is that?